packages/token-sdk/src/token-fetcher.ts

```diff
@@ -42,29 +46,27 @@ export class TokenFetcher {
     const mint = AddressUtil.toPubKey(address);
     const mintString = mint.toBase58();
     if (refresh || !this.contains(mintString)) {
-      const mintInfo = await this.request(
-        getParsedAccount(this.connection, mint, ParsableMintInfo)
-      );
-      invariant(mintInfo, "Mint info not found");
-      this._cache[mintString] = {
+      const mintInfo = await this.request(this.accountFetcher.getAccount(mint, ParsableMintInfo));
+      invariant(mintInfo, "Mint not found");
+      this._cache.set(mintString, {
         mint: mintString,
         decimals: mintInfo.decimals,
-      };
+      });
 
       for (const provider of this.providers) {
         try {
           const metadata = await this.request(provider.find(address));
-          this._cache[mintString] = mergeMetadata(this._cache[mintString], metadata);
+          this._cache.set(mintString, mergeMetadata(this._cache.get(mintString)!, metadata));
```

Comment:
nit: can we store & referenc the object saved in cache[mintString] in this block instead of hardcasting with !?